python/tk_framework_desktopserver/shotgun/__init__.py

```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+# We want class-level variables to persist, so we'll setup a cache here
+# that the factory functuon below populates.
```

Comment:
Rearranged some logic and it works fine now with global imports of the api_v1 and api_v2 modules. The module-level variable has been removed as of the next push.